jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1429,6 +1429,14 @@ public int compareTo(String string) {
 		return s1len - s2len;
 	}
 
+	private int compareValue(int codepoint) {
+		if ('A' <= codepoint && codepoint <= 'Z') {
+			return codepoint + ('a' - 'A');
+		}
+
+		return Character.toLowerCase(Character.toUpperCase(codepoint));
+	}
+
 	private char compareValue(char c) {
 		if (c < 128) {
```

Comment:
I expect the compareValue(int codepoint) code should be the same logic as compareValue(char c). It should either compare with 128, or the check against 128 should be removed here. 
@fjeremic  any JIT opinions?

Reply:
Int is signed vs char which is unsigned, so you'd have to check greater than 0 too, at which point you have two checks already. That was my reasoning for removing it for `compareValue(int codepoint)`.

Reply:
I think we should remove the `if (c < 128)` check in this API. The JIT will fold it away anyways since the `if ('A' <= c && c <= 'Z') {` is a stronger condition, but the method size will be inflated to the inliner, so getting rid of it in code is the best thing we can do.